### DISCUSSION TITLE
Failed to connect to GVRET at that com port

### DISCUSSION THREAD

**Discussion by jannerboyjohn** [collin80/SavvyCAN#868] [General]
Guys, I have the ESP32 board connected to the laptop with the Windows driver for Silicon Labs installed to recognise the device and com port. Installed the ESP32RET_Updater firmware ok. I am at my wits end with this trying to get the device to connect to SavvyCan. I use the Serial Connection GVRET. Doesn't matter what I try just getting the following type of error and continously scrolling data in the device console window.
Connecting to GVRET Device!
Write to serial -> e7 e7 f1 c f1 6 f1 7 f1 1 f1 9 
Failed to connect to GVRET at that com port
Serial connection to a GVRET device
Created Serial Port Object
Trying Standard Serial Mode
Connecting to GVRET Device!
Write to serial -> e7 e7 f1 c f1 6 f1 7 f1 1 f1 9 
Failed to connect to GVRET at that com port

I must be missing something for this to work/connect? Hopefully I have ! :-)
Appreciate if someone could help me resolve this.

Cheers, John

**Comment by DenisK63** [comment]
Ive got the same issue. Im posting a screenshot. 
![image](https://github.com/user-attachments/assets/100b0846-a13f-487c-bb91-847f362acd87)

**Comment by casefarmer1** [comment]
I'm running into the same problem, have you found the solution?

**Comment by ph-wheels** [comment]
same error & behaviour for me and I also tried it on MacOS with same result, tested it on multiple ESP boards, these are all ESP-VROOM-32D or ESP-VROOM-32U boards with esp chip's D0WD V1 & D0WDQ6 V1 all of these have an Silabs CP2102 uart and the ESP32RET loads fine, any clue or suggestion to resolve this ? Many thanks for your feedback

**Reply by ph-wheels** [reply]
Only fair to report back that the two esp32 (38pin, vroom-32D) after all turned out to be FAULTY, one probably beyond repair while the second one I could repair (bad 5V diode) and was re-used to test, this time after flashing it with esp32ret updater it did connect within savvycan perfectly, just to be sure I did also a full re-install of the cp2102 chip driver (which not really my favorite serial chip but it works)
and all seems to be fine again, hope this helps someone else

**Comment by MotorvateDIY** [comment]
Just use WiFi.

**Reply by meowmachine** [reply]
what do you mean? ESP32RET_Updater doesn't use wifi as far as I know. It turns your ESP32 into a serial device... Can you elaborate?

**Reply by meowmachine** [reply]
Nevermind. I see it's supposed to create a wifi AP but mine doesn't for some reason. Not sure why.

**Comment by ph-wheels** [comment]
Ok, thanks, with opensource wifi dongle van you adviseur
    
    Verzonden vanuit Outlook voor iOS
  

 Van: Frank Motorvate ***@***.***>Verzonden: dinsdag, februari 4, 2025 2:38 PMAan: collin80/SavvyCAN ***@***.***>CC: Paul ***@***.***>; Comment ***@***.***>Onderwerp: Re: [collin80/SavvyCAN] Failed to connect to GVRET at that com port (Discussion #868) 
Just use WiFi.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
Verzonden vanuit Outlook voor iOS

**Comment by meowmachine** [comment]
I also have the same issue. I've been trying to figure this out for weeks. Please help.

I am using the ESP-WROOM-32. I was able to flash the device using ESP32RET_Updater. I have tried flashing it from different computers and using Savvycan from different computers but no luck. 

![gvretnotworking](https://github.com/user-attachments/assets/4629737b-4e62-4269-9f9c-b6b5f4165e24)

**Comment by casefarmer1** [comment]
As before said, I had the same issue. Butt I fixed it now. After trying a lot of ESP32's and different PC's, decided to buy a ESP32 that was used in the yt video of Ali Bro. That works directly! The one thats works: https://nl.aliexpress.com/item/1005006792285064.html?spm=a2g0o.order_list.order_list_main.16.144979d2ME9FEN&gatewayAdapt=glo2nld

**Comment by make7535** [comment]
"ESP32RET
Reverse Engineering Tool running on ESP32 based hardware. **Supports both EVTV ESP32 and Macchina A0**
There is a precompiled binary version of this program here: https://www.savvycan.com/ESP32RET_Updater.zip"
Perhaps you have not configured the modules as required. EVTV ESP32 and Macchina A0

**Reply by ph-wheels** [reply]
thanks
